### PR TITLE
Implement Cosmos db new client with connection string

### PR DESCRIPTION
**Pull request by dmytrohridin**
Resolved https://github.com/Azure/azure-sdk-for-go/issues/18520

### CODE REVIEW

File: sdk/data/azcosmos/cosmos_client.go [hunk 1]
```diff
@@ -49,6 +50,42 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	return &Client{endpoint: endpoint, pipeline: newPipeline(newCosmosBearerTokenPolicy(cred, scope, nil), o)}, nil
 }
 
+// NewClientWithConnectionString creates a new instance of Cosmos client with connection string. It uses the default pipeline configuration.
+// connectionString - The cosmos service connection string.
+// options - Optional Cosmos client options.  Pass nil to accept default values.
+func NewClientWithConnectionString(connectionString string, o *ClientOptions) (*Client, error) {
```
Comment (ealsur) [inline sdk/data/azcosmos/cosmos_client.go:56]: Please update the doc.go file: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/doc.go#L27-L33

Comment (ealsur) [inline sdk/data/azcosmos/cosmos_client.go:56]: Please add a line in the Features Added section for 0.3.2: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/CHANGELOG.md#features-added

**Comment by dmytrohridin** [comment]
> Thanks for this! Could you please address the comments regarding docs/examples?

Sure, will add it later

**Comment by ealsur** [comment]
@jhendrixMSFT Could you verify from the API perspective this makes sense?

**Comment by dmytrohridin** [comment]
@ealsur @jhendrixMSFT 
done. please, recheck

**Review by jhendrixMSFT — APPROVED** [review]
API looks good.  Will defer to @ealsur for merging.

**Pull request merged**